packages/smithy-http/src/smithy_http/aio/crt.py

```diff
@@ -273,13 +273,15 @@ async def _get_connection(
     ) -> "crt_http.HttpClientConnection":
         # TODO: Use CRT connection pooling instead of this basic kind
         connection_key = (url.scheme, url.host, url.port)
-        if connection_key in self._connections:
-            return self._connections[connection_key]
-        else:
-            connection = await self._create_connection(url)
-            self._connections[connection_key] = connection
+        connection = self._connections.get(connection_key)
```

Comment:
What was prompting these changes?

Reply:
When using the `IMDSCredentialsResolver` with a `AWSCRTHTTPClient` as the configured client, I was running into the following error:

```
RuntimeError: 2058 (AWS_ERROR_HTTP_CONNECTION_CLOSED): The connection has closed or is closing.
```

The problem was that initial IMDS requests for getting the token will add a connection to the connection pool. This connection gets closes and remains in the pool. When we make the subsequent requests to get the iam profile, the error above will be throw since the cached connection is closed.